Description: Fix so we link against -ldl/-lssl etc
 The statically linked libmariadbclient.a has dependencies to -lssl and
 other dynamic libraries. Make sure those get added as well.
Author: Walter Doekes (work) <wjdoekes@osso.nl>
Forwarded: no
Last-Update: 2018-02-27

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -241,8 +241,9 @@ CONFIGURE_FILE(${CMAKE_SOURCE_DIR}/maria
 ####### MAODBC #######             
 ADD_LIBRARY(${LIBRARY_NAME} SHARED ${MARIADB_ODBC_SOURCES} ${CMAKE_SOURCE_DIR}/mariadb-odbc-driver-uni.def maodbcu.rc)
 
+MESSAGE(STATUS "Link dependencies: ${MARIADB_CLIENT_DEPENDENCIES}")
 SET_TARGET_PROPERTIES(maodbc PROPERTIES LANGUAGE C)
-TARGET_LINK_LIBRARIES(maodbc ${MARIADB_LIB} ${ODBC_INSTLIBS} ${PLATFORM_DEPENDENCIES})
+TARGET_LINK_LIBRARIES(maodbc ${MARIADB_LIB} ${ODBC_INSTLIBS} ${MARIADB_CLIENT_DEPENDENCIES} ${PLATFORM_DEPENDENCIES})
 
 # Currently on Windows only
 IF(WIN32)
--- a/cmake/FindMariaDB.cmake
+++ b/cmake/FindMariaDB.cmake
@@ -68,6 +68,19 @@ IF(NOT MARIADB_LIBRARY_DIR)
   ENDIF()
 ENDIF()
 
+IF(NOT WIN32 AND NOT MARIADB_CLIENT_DEPENDENCIES)
+  FIND_PATH(MARIADB_BIN_DIR mariadb_config
+            $ENV{MARIADB_DIR}/bin
+            ${MARIADB_DIR}/bin)
+  IF(MARIADB_BIN_DIR)
+    # we don't need the -lmariadb --lib, but we do need the other libs
+    # that the shared/static library requires
+    EXEC_PROGRAM(${MARIADB_BIN_DIR}/mariadb_config
+                 ARGS "--libs_sys"
+                 OUTPUT_VARIABLE MARIADB_CLIENT_DEPENDENCIES)
+  ENDIF()
+ENDIF()
+
 IF(MARIADB_LIBRARY_DIR AND MARIADB_INCLUDE_DIR)
   MESSAGE(STATUS "Found MariaDB libraries: ${MARIADB_LIBRARY_DIR}")
   SET(MARIADB_FOUND TRUE)
